src/tests/baseservices/invalid_operations/ManagedPointers.cs

```diff
@@ -0,0 +1,53 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+using Xunit;
+
+public unsafe class ManagedPointers
+{
+    [Fact]
+    public static void Validate_BoxingHelpers_NullByRef()
+    {
+        Console.WriteLine($"Running {nameof(Validate_BoxingHelpers_NullByRef)}...");
+        Assert.Throws<NullReferenceException>(() =>
+        {
+            object boxed = Unsafe.NullRef<int>();
+        });
+        Assert.Throws<NullReferenceException>(() =>
+        {
+            object boxed = Unsafe.NullRef<Guid>();
+        });
+        Assert.Throws<NullReferenceException>(() =>
+        {
+            object boxed = Unsafe.NullRef<string>();
+        });
+    }
+
+    [Fact]
+    public static void Validate_GeneratedILStubs_NullByRef()
+    {
+        Console.WriteLine($"Running {nameof(Validate_GeneratedILStubs_NullByRef)}...");
+        {
+            var fptr = (delegate*unmanaged<ref int, nint>)(delegate*unmanaged<void*, nint>)&PassByRef;
+            Assert.Equal(0, fptr(ref Unsafe.NullRef<int>()));
+        }
+
+        {
+            var fptr = (delegate*unmanaged<ref Guid, nint>)(delegate*unmanaged<void*, nint>)&PassByRef;
+            Assert.Equal(0, fptr(ref Unsafe.NullRef<Guid>()));
+        }
+
+        Assert.Throws<NullReferenceException>(() =>
+        {
+            var fptr = (delegate*unmanaged<ref string, nint>)(delegate*unmanaged<void*, nint>)&PassByRef;
+            fptr(ref Unsafe.NullRef<string>());
+        });
+
+        [UnmanagedCallersOnly]
+        static nint PassByRef(void* a) => (nint)a;
+    }
```

Comment:
There are other interesting APIs that take byrefs. For example: `Interlocked.CompareExchange`. If I am reading the code correctly, Mono is not handling the null byref passed into `Interlocked.CompareExchange`.

Reply:
Thanks. Looking for these troublesome APIs is what I was going to look for tomorrow.

Reply:
Looking at the mono implementation I agree it looks like it would fail. They do run and pass on Windows x64 though. Going to see how the other platforms fare.

/cc @lambdageek

Reply:
Fails on OSX as expected.